### ISSUE TITLE
Validation for registry credentials still not working as expected

### ISSUE THREAD

**Issue by TNierman** [metacloud/molecule#1330]
# Issue Type

- Bug report

# Molecule and Ansible details

```
ansible --version
ansible 2.5.4
  config file = /home/Trevor.Nierman/.ansible.cfg
  configured module search path = [u'/home/Trevor.Nierman/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /home/Trevor.Nierman/venv_ansible-2.5.2/local/lib/python2.7/site-packages/ansible
  executable location = /home/Trevor.Nierman/venv_ansible-2.5.2/bin/ansible
  python version = 2.7.15rc1 (default, Apr 15 2018, 21:51:34) [GCC 7.3.0]

```

```
molecule --version
molecule, version 2.14.0
```

Molecule installation method:

- pip

Ansible installation method:

- pip

# Desired Behavior

Issue first reported here: https://github.com/metacloud/molecule/issues/1297
Molecule should support the use of private docker registries, as previously mentioned. For example:

```
platforms:
  - name: <container_name>
    image: "<docker_image>"
    registry:
      url: <docker_registry_url>
      credentials:
        username: "$DOCKER_REGISTRY_USER"
        password: "{$DOCKER_REGISTRY_PASS}"
```

(For the sake of clarity, <all_lower> denotes fields that can be plaintext, $ALL_CAPS denotes env vars)
Currently, this feature requires that username and password be passed as env vars.

# Actual Behaviour (Bug report only)

```
$ molecule --debug create
--> Validating schema ###/molecule.yml.
ERROR: Failed to validate.

{'platforms': [{0: [{'registry': [{'credentials': [{'password': ["value does not match regex '^[{$]+[a-z0-9A-z]+[}]*$'"]}]}]}]}]}
```

Changing the format in any way, ie: 
```
        credentials:
        username: "$DOCKER_REGISTRY_USER"
        password: {"$DOCKER_REGISTRY_PASS"}
```

 results in:
```
$ molecule --debug create
ERROR: Failed to validate.

{'platforms': [{0: [{'registry': [{'credentials': [{'password': ['must be of string type']}]}]}]}]}
```

**Comment by TNierman** [comment]
Whoops, link didn't copy correctly!

Issue first reported here: https://github.com/metacloud/molecule/issues/1297

**Comment by TNierman** [comment]
Whoops, looks like user error. 

Credentials should be written as: 
```
credentials:
    username: "$USER"
    password: {"PASSWORD"}
```

**Comment by retr0h** [comment]
Not sure I follow, but we should only allow ${VAR} or $VAR.  The regexp looks wrong.  Not sure what you were trying to accomplish with {"PASSWORD"} but that is deff wrong.  Molecule should not be allowing that.

**Comment by retr0h** [comment]
This bug was reported and fixed in #1315.  However, 2.15 is not released yet.
